### PR TITLE
Long running query

### PR DESCRIPTION
**Pull request by SylviaZiyuZhang**
# What

# Background

# Test Plan

### CODE REVIEW

File: driver/collector/postgres_collector.py [hunk 1]
```diff
@@ -320,6 +320,29 @@
...
 """
 
+# query digest id not included pre pg 14
+LONG_RUNNING_QUERY_NO_ID_SQL_TEMPLATE = """
+SELECT
+    pid, query_start, datid, datname, usename, state
+FROM
+    pg_stat_activity
+WHERE
+    now() - query_start > interval '{latency_threshold_min} minutes';
```
Comment (YuzeLiao) [inline driver/collector/postgres_collector.py:330]: Maybe we can calculate a timestamp like "earliest_time", and do:

`WHERE query_start > {earlist_time}` ?

File: driver/collector/postgres_collector.py [hunk 2]
```diff
@@ -320,6 +320,29 @@
...
+    {n};
+"""
+
+LONG_RUNNING_QUERY_SQL_TEMPLATE = """
+SELECT
+    pid, query_id, query_start, datid, datname, usename, state
+FROM
+    pg_stat_activity
+WHERE
+    now() - query_start > interval '{latency_threshold_min} minutes';
```
Comment (YuzeLiao) [inline driver/collector/postgres_collector.py:341]: Same above

**Review by YuzeLiao — APPROVED** [review]

**Review by bohanjason — APPROVED** [review]

File: driver/collector/postgres_collector.py [hunk 3]
```diff
@@ -320,6 +322,29 @@
...
+    pg_stat_activity
+WHERE
+    query_start < {lr_query_start_timestamp};
+LIMIT
+    {n};
+"""
+
+LONG_RUNNING_QUERY_SQL_TEMPLATE = """
+SELECT
+    pid, query_id, query_start, datid, datname, usename, state
```
Comment (bohanjason) [inline driver/collector/postgres_collector.py:339]: I don't think we need to collect username here.

We also need these fields: wait_event, backend_type, xact_start, state_change

**Review by bohanjason — CHANGES_REQUESTED** [review]

File: driver/collector/postgres_collector.py (hunk 3 above)
Comment (bohanjason) [reply driver/collector/postgres_collector.py:339]: wait_event_type

File: driver/config/driver_config.yaml [hunk 4]
```diff
@@ -24,6 +24,7 @@ monitor_interval: 60
 table_level_monitor_interval: 3600
 num_table_to_collect_stats: 1000
 num_index_to_collect_stats: 10000
+long_running_query_monitor_interval: 240
```
[the remark this answers is not in the document]
Comment (SylviaZiyuZhang) [reply driver/config/driver_config.yaml:27]: Added new args for disabling

File: driver/collector/postgres_collector.py (hunk 3 above)
Comment (SylviaZiyuZhang) [reply driver/collector/postgres_collector.py]: Added the proposed fields

**Review by bohanjason — APPROVED** [review]
LGTM, thanks!

**Pull request merged**